crates/primitives/src/env.rs

```diff
@@ -92,6 +92,25 @@ impl Env {
     /// Return initial spend gas (Gas needed to execute transaction).
     #[inline]
     pub fn validate_tx<SPEC: Spec>(&self) -> Result<(), InvalidTransaction> {
+        // Check if the transaction's chain id is correct
+        if let Some(tx_chain_id) = self.tx.chain_id {
+            if tx_chain_id != self.cfg.chain_id {
```

Comment:
per the spec: 

>Verify the chain id is either 0 or the chain’s current ID.

so we should also check if the chain id is 0 for 7702 tx's

Reply:
That is for `authorization_list` where `chain_id` can be zero.

The transaction chain_id needs to be checked, and it is invalid if it is zero for Ethereum